### PR TITLE
Backward compatibility when using new SearchPath APIs

### PR DESCRIPTION
**Pull request by dennis-behm**
When introducing the usage of the new SearchPath APIs, backward compatibility was broken for some language scripts. 

This PR
* fixes the backward compatibility and
* introduces assertion of the requires `*langPrefix*_dependencySearch` configuration

### CODE REVIEW

File: test/applications/MortgageApplication/application-conf/Cobol.properties [hunk 1]
```diff
@@ -10,6 +10,11 @@ cobol_fileBuildRank=
 # Rules defined in application.properties
 cobol_resolutionRules=[${copybookRule}]
 
+#
+# COBOL dependencySearch configuration
+# searchPath defined in application.properties
+cobol_dependencySearch=${copybookSearch}
```
Comment (lburgess07) [inline test/applications/MortgageApplication/application-conf/Cobol.properties:16]: I thought this was in previous versions of zAppBuild? Why is git showing it like this is a new addition to the file?

File: test/applications/MortgageApplication/application-conf/Cobol.properties [hunk 2]
```diff
@@ -59,6 +64,12 @@ cobol_linkDebugExit=    INCLUDE OBJECT(@{member})  \n    INCLUDE SYSLIB(EQAD3CXT
 # can be overridden by file properties
 cobol_linkEdit=true
 
+#
+# store abbrev git hash in ssi field
+# available for buildTypes impactBuild, mergeBuild and fullBuild 
+# can be overridden by file properties
+cobol_storeSSI=true
```
Comment (lburgess07) [inline test/applications/MortgageApplication/application-conf/Cobol.properties:71]: Same question here. I did not make these changes on my branch - are they intentional?

**Review by lburgess07 — COMMENTED** [review]
Looks good overall - I have two comments about some additions displayed for Cobol.properties.

File: test/applications/MortgageApplication/application-conf/Cobol.properties (hunk 2 above)
Comment (nixpix49) [reply test/applications/MortgageApplication/application-conf/Cobol.properties:71]: @lburgess07 Noticed that the changes are in test/applications/... ?

**Review by lburgess07 — APPROVED** [review]

**Comment by dennis-behm** [comment]
This PR needs to flow into the zAppBuild_2_x stream. Therefore I changed the base branch to target the correct stream.

**Pull request merged**